added subject types docs (#37075)
DataDog/documentation: content/en/experiments/subject_types.md

---
title: Manage Subject Types
description: Configure subject types so Datadog can randomize experiments and join exposures to metrics.
further_reading:
- link: "/experiments/plan_and_launch_experiments"
  tag: "Documentation"
  text: "Plan and Launch Experiments"
- link: "/experiments/defining_metrics/?tab=warehouse"
  tag: "Documentation"
  text: "Create Experiment Metrics From Warehouse Data"
- link: "/experiments/reading_results"
  tag: "Documentation"
  text: "Read Experiment Results"
---

## Overview

> Use this page to configure **subject types**—the unit at which Datadog randomizes experiment traffic (for example, a user or an organization).

Reviewer: ```suggestion
Use this page to configure subject types, the unit Datadog uses to randomize experiment traffic, such as a user or an organization.
```
Suggest to reword a bit for clarity and remove the em dash

> Experiments assign traffic to different variations of an experience and compare outcomes. For most use cases, subjects correspond to users, but you can define other units of randomization.

Reviewer: ```suggestion
Experiments assign traffic to different variations of an experience and measure outcomes across those groups. Subjects are users by default, but you can define other randomization units.
```

> For a given subject, the Datadog SDK returns the same variant on every evaluation. This helps keep the experience consistent across sessions and devices.

Reviewer: ```suggestion
For a given subject, the Datadog SDK returns the same variant on every evaluation, keeping the experience consistent across sessions and devices.
```
same with this, moving to a single sentence to tighten up the flow.

Subject types also determine how Datadog associates metric events with experiment exposures for statistical analysis.

## Specify the subject identifier in code

When you use the Datadog SDK, pass a unique identifier for the experiment subject as the `targetingKey`:

```js
const evaluationContext = {

>     targetingKey: user_id,

Reviewer: `user_id` is a bare identifier, the example later in uses `'user_1234'`. Suggest to match that pattern for consistency. But please correct me if it should just be `user_id` here :) 

```suggestion
    targetingKey: 'user_1234',
```

};
```

Randomization uses deterministic hashing. Each time the user returns to the application, Datadog assigns the same variant as long as the user remains eligible based on targeting rules.

## Map experiment exposures to metrics

> When you [plan an experiment][1], select the subject type above the primary metric:

Reviewer: "Above" is directional language that depends on the reader seeing the UI as we assume they would. Per AI-readability guidance, we should prefer functional descriptions over positional ones. Consider rephrasing or dropping "above the primary metric" entirely, since the screenshot that follows shows the layout.

{{< img src="/product_analytics/experiment/subject_types/select_subject_type.png" alt="The experiment setup page with the subject type selector above primary metric selection, showing User and Organization options." style="width:80%;" >}}

By default, Datadog includes a **User** subject type. To manage subject types, navigate to **Experiments** > **Settings** > **Subject Types**:

{{< img src="/product_analytics/experiment/subject_types/subject_type_list.png" alt="The Subject Types settings page listing User and Organization subject types." style="width:80%;" >}}

### Warehouse metrics

When you configure [Metric SQL Models][2], map warehouse columns to subject types in the right panel:

{{< img src="/product_analytics/experiment/subject_types/sql_model_setup.png" alt="The Metric SQL Model setup page with warehouse columns mapped to subject types in the right panel." style="width:80%;" >}}

<div class="alert alert-info">If you add a default warehouse column ID on the Subject Types page (for example, <code>user_id</code>), Datadog automatically maps columns with that name to the matching subject type.</div>

> Datadog uses the subject column to join experiment exposure data in the `g_exposures` table to your metric SQL. Make sure values in the subject column match the identifiers you pass as `targetingKey` in the SDK.

Reviewer: `g_exposures` appears here without context, so the reader has no way to know what this table is. Suggest to briefly explain on first mention (for example, "the `g_exposures` table, which records each experiment assignment") or link to a reference.

---
title: Manage Subject Types
description: Configure subject types so Datadog can randomize experiments and join exposures to metrics.
further_reading:
- link: "/experiments/plan_and_launch_experiments"
  tag: "Documentation"
  text: "Plan and Launch Experiments"
- link: "/experiments/defining_metrics/?tab=warehouse"
  tag: "Documentation"
  text: "Create Experiment Metrics From Warehouse Data"
- link: "/experiments/reading_results"
  tag: "Documentation"
  text: "Read Experiment Results"
- link: "/experiments/guide/connecting_a_data_warehouse/"
  tag: "Documentation"
  text: "Connect your data warehouse"
---

## Overview

> Use this page to configure subject types, the unit Datadog uses to randomize experiment traffic, such as a user or an organization.

Reviewer: ```suggestion
Use this page to configure subject types—the unit Datadog uses to randomize experiment traffic—such as a user or an organization.
```
ok I changed my mind and going back to the "em" dash here, I think it lands better! Sorry :)
